gladier_xpcs/tools/gather_xpcs_metadata.py

```diff
@@ -152,12 +152,12 @@ def gather(dataframe):
         # Parent: sanat
         metadata['parent'] = re.search(r'([a-z]+)*', root_folder.name).group()
     except Exception:
-        p = pathlib.Path(event['proc_dir']) / 'gather_xpcs_metadata.error'
+        p = pathlib.Path(data['proc_dir']) / 'gather_xpcs_metadata.error'
         with open(str(p), 'w+') as f:
             f.write(traceback.format_exc())
 
 
-    pilot = event['pilot']
+    pilot = data['pilot']
     # metadata passed through from the top level takes precedence. This allows for
     # overriding fields through $.input
     metadata.update(pilot.get('metadata', {}))
```

Comment:
Want to add the metadata from below under this line? 

```
    try:
        import boost_corr
        boost = True
        boost_version = boost_corr.__version__
        gpu = data.get('gpu_flag','0')
    except:
        boost = False
        boost_version = 0
        gpu = None
```

I think it would be good to generalize this under a small object like executable, such as: 
```
metadata.update({
' executable' : {
  'name': 'boost_corr'
  'version': version,
  'device: 'gpu',
  'source': 'https://pypi.org/project/boost_corr/',
}
})
metadata.update(pilot.get('metadata', {}))
```